### PR TITLE
conn_handler: fallback to strtod if needed

### PR DESCRIPTION
**Pull request by filippog**
namely to support scientific notation, see also #123

### CODE REVIEW

File: SConstruct [hunk 1]
```diff
@@ -6,8 +6,8 @@ murmur = envmurmur.Library('murmur', Glob("deps/murmurhash/*.c"))
 envinih = Environment(CPATH = ['deps/inih/'], CFLAGS="-O3")
 inih = envinih.Library('inih', Glob("deps/inih/*.c"))
 
-env_statsite_with_err = Environment(CCFLAGS = '-g -std=c99 -D_GNU_SOURCE -Wall -Werror -Wstrict-aliasing=0 -O3 -pthread -Ideps/inih/ -Ideps/libev/ -Isrc/')
-env_statsite_without_err = Environment(CCFLAGS = '-g -std=c99 -D_GNU_SOURCE -O3 -pthread -Ideps/inih/ -Ideps/libev/ -Isrc/')
+env_statsite_with_err = Environment(CCFLAGS = '-g -std=c99 -D_GNU_SOURCE -D_REENTRANT -Wall -Werror -Wstrict-aliasing=0 -O3 -pthread -Ideps/inih/ -Ideps/libev/ -Isrc/')
```
Comment (armon) [inline SConstruct]: I don't think this is required. Setting `-pthread` implies it already

File: src/conn_handler.c [hunk 2]
```diff
@@ -444,7 +451,9 @@ static int handle_ascii_client_connect(statsite_conn_handler *handle) {
 
         // Convert the value to a double
         val = str2double(val_str, &endptr);
-        if (unlikely(endptr == val_str)) {
+        if (unlikely(endptr == val_str ||
+              (errno == ERANGE &&
+                 (val == HUGE_VAL || val == -HUGE_VAL || val == 0)))) {
```
Comment (armon) [inline src/conn_handler.c]: I don't think we even need to check `val` if `errno == ERANGE` that is enough to error out.

File: SConstruct (hunk 1 above)
Comment (filippog) [inline SConstruct]: you are right, dropped the commit

File: src/conn_handler.c (hunk 2 above)
Comment (filippog) [inline src/conn_handler.c]: indeed, I've dropped the additional checks and reset `errno` even in the non-strtod case

**Comment by armon** [comment]
@filippog Could you add tests to ensure that the metrics in the given formats are correctly parsed?

**Comment by filippog** [comment]
@armon yep, ideally I'd like to unit test str2double but how to do that cleanly with static functions? `#include "conn_handler.c"` in the test runner doesn't feel right, perhaps moving `str2double` to e.g. `src/utils.[ch]`

**Comment by armon** [comment]
@filippog It should be enough to handle this with integration tests. It is annoying to unit test it as is, and it seems weird to export it.

**Pull request merged**